Source/Core/Core/AchievementManager.cpp

```diff
@@ -111,4 +123,210 @@ AchievementManager::ResponseType AchievementManager::Request(
   }
 }
 
+void AchievementManager::EnableDLL(bool enable)
+{
+  m_dll_enabled = enable;
+}
+
+bool AchievementManager::IsDLLEnabled()
+{
+  return m_dll_enabled;
+}
+
+#ifdef ENABLE_RAINTEGRATION
+void AchievementManager::InitializeRAIntegration(void* main_window_handle)
+{
+  if (!m_dll_enabled)
+    return;
+  RA_InitClient((HWND)main_window_handle, "Dolphin", SCM_DESC_STR);
+  RA_SetUserAgentDetail(std::format("Dolphin {} {}", SCM_DESC_STR, SCM_BRANCH_STR).c_str());
+
+  RA_InstallSharedFunctions(RACallbackIsActive, RACallbackCauseUnpause, RACallbackCausePause,
+                            RACallbackRebuildMenu, RACallbackEstimateTitle, RACallbackResetEmulator,
+                            RACallbackLoadROM);
+
+  // EE physical memory and scratchpad are currently exposed (matching direct rcheevos
+  // implementation).
+  m_memory_manager = &Core::System::GetInstance().GetMemory();
+  ReinstallMemoryBanks();
+
+  m_raintegration_initialized = true;
+
+  RA_AttemptLogin(0);
+
+  // this is pretty lame, but we may as well persist until we exit anyway
+  std::atexit(RA_Shutdown);
+}
+
+void AchievementManager::ReinstallMemoryBanks()
+{
+  if (!m_dll_enabled)
+    return;
+  RA_ClearMemoryBanks();
+  int memory_bank_size = 0;
+  if (Core::GetState() != Core::State::Uninitialized)
+  {
+    memory_bank_size = m_memory_manager->GetExRamSizeReal();
+  }
+  RA_InstallMemoryBank(0, RACallbackReadMemory, RACallbackWriteMemory, memory_bank_size);
+  RA_InstallMemoryBankBlockReader(0, RACallbackReadBlock);
+}
```

Comment:
Uh... explain to me what this is doing, because I suspect you don't just want to access MEM2...? `ExRam` is the Wii's additional memory.